crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,1418 @@
+pub unsafe extern "C" fn qk_target_min_length(target: *const Target) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.min_length
+}
+
+/// @ingroup QkTarget
+/// Returns the `pulse_alignment` value of this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return The pulse_alignment value of this Target.
+///
+/// # Example
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     // The value defaults to 1
+///     uint32_t pulse_alignment = qk_target_pulse_alignment(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_pulse_alignment(target: *const Target) -> u32 {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.pulse_alignment
+}
+
+/// @ingroup QkTarget
+/// Returns the `acquire_alignment` value of this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return The acquire_alignment value of this Target.
+///
+/// # Example
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     // The value defaults to 0
+///     uint32_t acquire_alignment = qk_target_pulse_alignment(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_acquire_alignment(target: *const Target) -> u32 {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.acquire_alignment
+}
+
+/// @ingroup QkTarget
+/// Sets the description of this Target.
+///
+/// @param target A pointer to the Target.
+/// @param description A string describing the Target.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     qk_target_set_description(target, "New_Name");
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+///
+/// The ``description`` type is expected to be a pointer to a null-terminated valid string. If the
+/// pointer is not valid or not null ending, the behavior will be undefined.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_set_description(
+    target: *mut Target,
+    description: *mut c_char,
+) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { mut_ptr_as_ref(target) };
+    // SAFETY: Per documentation, name points to a null-terminated string of characters.
+    let description: &CStr = unsafe { CStr::from_ptr(description) };
+    target.description = Some(description.to_str().unwrap().to_string());
+    ExitCode::Success
```

Comment:
Since this always either returns `Success` or fails I don't think we need an exit code here 🤔  but I'm also fine keeping it if you really want to, but then it should add it to the docs 🙂 

This comment also applies to some other functions below.